src/app/api/users/users.service.ts

```diff
@@ -0,0 +1,30 @@
+import { CopilotAPI } from '@/utils/CopilotAPI'
+import { BaseService } from '@api/core/services/base.service'
+import User from '@api/core/models/User.model'
+import { PoliciesService } from '@api/core/services/policies.service'
+import { Resource } from '@api/core/types/api'
+import { UserAction } from '@api/core/types/user'
+
+class UsersService extends BaseService {
+  private copilot: CopilotAPI
+
+  constructor(user: User) {
+    super(user)
+    this.copilot = new CopilotAPI(this.user.token)
+  }
+
+  async getGroupedUsers() {
+    const user = this.user
+    new PoliciesService(user).authorize(UserAction.Read, Resource.Users)
+
+    const [ius, clients, companies] = await Promise.all([
+      this.copilot.getInternalUsers(),
+      this.copilot.getClients(),
```

Comment:
What if there are thousands of clients in a portal? Is this going to fetch through all pages? Who will this be handled?

Reply:
That is definitely an issue as we scale up. This endpoint is meant to fetch users for this view:

![image](https://github.com/copilot-platforms/tasks-app/assets/59971845/a8b85580-bc0c-4387-956d-ec9ba8e4c4f4)

So if we're discussing pagination we should figure out how we will scale up this view dropdown when there are lots of users as well. How would we handle this in the design?

I will create a new issue in backlog for this for now

Reply:
I think this design will still scale to filter assignees, but the search on the drop-down will have to be done through an api call rather than just filtering on the webapp. wdyt?

Reply:
Makes sense. Currently we are doing a client-side filter for this which won't scale well.

I've made a linear issue for this, we will look over this in the future.